src/transcript.rs

```diff
@@ -39,30 +41,36 @@ pub trait TranscriptWrite<C: CurveAffine>: Transcript<C> {
     fn write_scalar(&mut self, scalar: C::Scalar) -> io::Result<()>;
 }
 
-/// This is just a simple (and completely broken) transcript reader
-/// implementation, standing in for some algebraic hash function that we'll
-/// switch to later.
+/// We will replace BLAKE2b with an algebraic hash function in a later version.
 #[derive(Debug, Clone)]
-pub struct DummyHashRead<R: Read, C: CurveAffine> {
-    base_state: C::Base,
-    scalar_state: C::Scalar,
+pub struct Blake2bRead<R: Read, C: CurveAffine> {
+    base_state: Blake2bState,
+    scalar_state: Blake2bState,
     read_scalar: bool,
     reader: R,
+    _marker: PhantomData<C>,
 }
 
-impl<R: Read, C: CurveAffine> DummyHashRead<R, C> {
+impl<R: Read, C: CurveAffine> Blake2bRead<R, C> {
     /// Initialize a transcript given an input buffer and a key.
-    pub fn init(reader: R, key: C::Base) -> Self {
-        DummyHashRead {
-            base_state: key + &C::Base::from_u64(1013),
-            scalar_state: C::Scalar::from_u64(1013),
+    pub fn init(reader: R) -> Self {
+        Blake2bRead {
+            base_state: Blake2bParams::new()
+                .hash_length(32)
+                .personal(C::BLAKE2B_PERSONALIZATION)
+                .to_state(),
+            scalar_state: Blake2bParams::new()
+                .hash_length(32)
+                .personal(C::BLAKE2B_PERSONALIZATION)
+                .to_state(),
```

Comment:
This impl can be simplified by having only one instances of a BLAKE2b hasher involved and hashing both the scalars and the group elements into the same hasher. The reason why there were separate hashers in the old implementation is that it was loosely describing an algebraic hash where we will indeed be forced to have separate hashers because they're separate fields.